graphdatascience/query_runner/aura_db_query_runner.py

```diff
@@ -120,32 +130,43 @@ def _remote_write_back(
         logging: bool = False,
         custom_error: bool = True,
     ) -> DataFrame:
-        if params["config"] is None:
-            params["config"] = {}
+        config = params["config"] if "config" in params else {}
 
         # we pop these out so that they are not retained for the GDS proc call
-        db_arrow_config = params["config"].pop("arrowConfiguration", {})  # type: ignore
+        db_arrow_config = config.pop("arrowConfiguration", {})
         self._inject_arrow_config(db_arrow_config)
 
-        job_id = params["config"]["jobId"] if "jobId" in params["config"] else str(uuid4())  # type: ignore
-        params["config"]["jobId"] = job_id  # type: ignore
+        job_id = config["jobId"] if "jobId" in config else str(uuid4())
+        config["jobId"] = job_id
 
-        params["config"]["writeToResultStore"] = True  # type: ignore
+        config["writeToResultStore"] = True
 
         gds_write_result = self._gds_query_runner.call_procedure(
             endpoint, params, yields, database, logging, custom_error
         )
 
-        db_write_proc_params = {
-            "graphName": params["graph_name"],
-            "databaseName": self._gds_query_runner.database(),
-            "jobId": job_id,
-            "arrowConfiguration": db_arrow_config,
-        }
+        graph_name = params["graph_name"]
+        if ProtocolVersion.V2 in self._protocol_versions:
+            db_write_proc_params = self._write_back_params_v2(graph_name, job_id, db_arrow_config, config)
+            protocol_version = ProtocolVersion.V2
+        elif ProtocolVersion.V1 in self._protocol_versions:
+            db_write_proc_params = self._write_back_params_v1(graph_name, job_id, db_arrow_config)
+            protocol_version = ProtocolVersion.V1
```

Comment:
I was expecting to see something similar to this for projections as well.

Reply:
Yes, you're right I confused a bit which of the projection procedures and aggregation functions is called in which order 🤷